modules/performance_analyzer.f90

```diff
@@ -43,44 +48,122 @@ subroutine get_hamiltonian_performance(lgflops)
+    call MPI_ALLREDUCE(tin,tout,1,MPI_DOUBLE_INT,MPI_MAXLOC,MPI_COMM_WORLD,ierr)
+    call MPI_BCAST(pgflops,4,MPI_REAL8,tout%rank,MPI_COMM_WORLD,ierr)
 
 #ifdef ARTED_MS
     call MPI_ALLREDUCE(lgflops,sgflops,4,MPI_REAL8,MPI_SUM,NEW_COMM_WORLD,ierr)
 #endif
     call MPI_ALLREDUCE(lgflops,tgflops,4,MPI_REAL8,MPI_SUM,MPI_COMM_WORLD,ierr)
 
     if(Myrank == 0) then
-      print *, '###'
-      print *, 'Performance [GFLOPS]'
-      print *, '  Processor'
-      print *, '    hamiltonian  :', lgflops(4)
-      print *, '    - stencil    :', lgflops(1)
-      print *, '    - pseudo pt. :', lgflops(2)
-      print *, '    - update     :', lgflops(3)
+      write (iounit,'(A)') 'Performance [GFLOPS]'
+      write (iounit,'(A,4(A15))') 'Type           ', 'Hamiltonian', 'Stencil', 'Pseudo-Pt', 'Update'
+      write (iounit,f)            'Processor      ', lgflops(4), lgflops(1), lgflops(2), lgflops(3)
+      write (iounit,f)            'Processor(max) ', pgflops(4), pgflops(1), pgflops(2), pgflops(3)
 #ifdef ARTED_MS
-      print *, '  Macroscopic grid point'
-      print *, '    hamiltonian  :', sgflops(4)
-      print *, '    - stencil    :', sgflops(1)
-      print *, '    - pseudo pt. :', sgflops(2)
-      print *, '    - update     :', sgflops(3)
+      write (iounit,f)            'Macro-grid(sum)', sgflops(4), sgflops(1), sgflops(2), sgflops(3)
 #endif
-      print *, '  System total'
-      print *, '    hamiltonian  :', tgflops(4)
-      print *, '    - stencil    :', tgflops(1)
-      print *, '    - pseudo pt. :', tgflops(2)
-      print *, '    - update     :', tgflops(3)
-      print *, '###'
-    endif
+      write (iounit,f)            'System(sum)    ', tgflops(4), tgflops(1), tgflops(2), tgflops(3)
+    end if
+  end subroutine
+
+  subroutine write_loadbalance(iounit)
+    use global_variables
+    use timelog
+    implicit none
+    integer,intent(in) :: iounit
+
+    integer,parameter      :: LOG_SIZE=12
+    character(*),parameter :: f = "(A,3(F12.4),F12.2)"
+
+    real(8) :: src(LOG_SIZE), rmin(LOG_SIZE), rmax(LOG_SIZE), diff(LOG_SIZE), rel(LOG_SIZE)
+
+    src( 1) = timelog_get(LOG_DT_EVOLVE)
+    src( 2) = timelog_get(LOG_HPSI)
+    src( 3) = timelog_get(LOG_PSI_RHO)
+    src( 4) = timelog_get(LOG_HARTREE)
+    src( 5) = timelog_get(LOG_CURRENT)
+    src( 6) = timelog_get(LOG_TOTAL_ENERGY)
+    src( 7) = timelog_get(LOG_ION_FORCE)
+    src( 8) = timelog_get(LOG_DT_EVOLVE_AC)
+    src( 9) = timelog_get(LOG_K_SHIFT_WF)
+    src(10) = timelog_get(LOG_OTHER)
+    src(11) = timelog_get(LOG_ALLREDUCE)
+    src(12) = timelog_get(LOG_DYNAMICS)
+
+    call MPI_ALLREDUCE(src,rmin,LOG_SIZE,MPI_REAL8,MPI_MIN,MPI_COMM_WORLD,ierr)
+    call MPI_ALLREDUCE(src,rmax,LOG_SIZE,MPI_REAL8,MPI_MAX,MPI_COMM_WORLD,ierr)
+
+    diff(:) = rmax(:) - rmin(:)
+    rel(:)  = rmax(:) / rmin(:)
+
+    if (Myrank == 0) then
+      write (iounit,'(A)') 'Load balance check [sec]'
+      write (iounit,'(A,4(A12))') 'Function    ','min','max','diff','rel [%]'
```

Comment:
Thank you for implementing the load balance checker! I have a comment. Probably, the value of "rel()" should be multiplied by a factor of 100 to be described as a percentage.